### ISSUE TITLE
Test: Add warning message to 'Expunge Deleted Titles' option

### ISSUE THREAD

**Issue by ostephens** [k-int/KBPlus#1020]
(Requested as critical for v5.0 by MB via email to OS 2015-08-19)

Add warning message popup triggered when user chooses ‘Expunge Deleted Titles’ option (https://www.kbplus.ac.uk/test/dataManager/expungeDeletedTitles)

Message to read “You are about to permanently delete all titles with a status of ‘Deleted’. This will also delete any TIPPs and IEs that are attached to this title. Only click OK if you are absolutely sure you wish to proceed.” Option to either go ahead “OK” or cancel “Cancel”

**Comment by iyawnis** [comment]
I assume a JS confirm will do ?

**Comment by iyawnis** [comment]
@ostephens What about expungeDeletedTIPPS? Any message for that ?

**Comment by ostephens** [comment]
@latusaki can you confirm what happens to any IEs that are linked to the TIPPs removed via expungeDeletedTIPPS?

**Comment by iyawnis** [comment]
I think the action fails due to constrain violation. Should the ies be
deleted

**Comment by iyawnis** [comment]
Yeah confirmed now. getting this error in console, processing continues and will work for tipps that dont have IEs 

```
    ... 14 more
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Cannot delete or update a parent row: a foreign key constraint fails (`kbpluslive`.`issue_entitlement`, CONSTRAINT `FK2D45F6C7330B4F5` FOREIGN KEY (`ie_tipp_fk`) REFERENCES `title_instance_package_platform` (`tipp_id`))
    at com.mysql.jdbc.Util.handleNewInstance(Util.java:409)
```

**Comment by ostephens** [comment]
I'm reluctant for this to delete IEs that are still being used.
I think my preference would be that IEs should only be deleted if they also have status Deleted, or they are linked to a Subscription with a status of Deleted.

Not sure how much this work is though. If this is difficult, then I'd suggest we leave it for that moment so that TIPPs attached to IEs are not deleted, with a warning message saying "This will only delete TIPPs that are not attached to existing IEs." OK to continue, Cancel to cancel

**Comment by ianibbo** [comment]
So have an IE linked to a title which is deleted?

not possible, would cause very bad things [TM] to happen

Ian Ibbotson
Director
Knowledge Integration Ltd
35 Paradise Street, Sheffield. S3 8PZ
T: 0114 273 8271
M: 07968 794 630
W: http://www.k-int.com

**Comment by ostephens** [comment]
Was in response to @latusaki question about expungeDeletedTIPPS - so not TI deletion in this case.
Happy to leave expungeDeletedTIPPS behaviour as is for v5.0 with addition of a pop-up warning "This will only delete TIPPs that are not attached to existing IEs." OK to continue, Cancel to cancel

**Comment by iyawnis** [comment]
@ostephens  I just added some logic that will delete IEs of deleted subs, or IEs with status deleted. For the rest we cant do anything but log the error, and the TIPP will not be removed. I will add the warning message you suggested as well.

**Comment by iyawnis** [comment]
In the future if necessary we can add a dedicated screen that will show any IEs that stop the deletion etc

**Comment by ostephens** [comment]
@latusaki thanks. In that case minor modification to message:
"This will only delete TIPPs that are not attached to current IEs."

**Comment by iyawnis** [comment]
Fixed here: 8d154b8c8cbe24061d3d3eca099f17e09acdacee 

Still need to delete IEs on sub deleted.

**Comment by latusaki** [comment]
Will mak
